queries/file.py

```diff
@@ -83,3 +83,30 @@ def construct_get_file_query(file_uri, graph=APPLICATION_GRAPH):
     return query_template.substitute(
         graph=sparql_escape_uri(graph),
         file_uri=sparql_escape_uri(file_uri))
+
+def delete_physical_file_metadata(file_uri):
+    """
+    Construct a SPARQL query for deleting the metadata of a physical file.
+    :param file_uri: string containing physical file uri
+    :returns: string containing SPARQL query
+    """
+    query_template = Template("""
+PREFIX nfo: <http://www.semanticdesktop.org/ontologies/2007/03/22/nfo#>
+PREFIX nie: <http://www.semanticdesktop.org/ontologies/2007/01/19/nie#>
+
+DELETE {
+    $file_uri ?p ?o .
+    ?logicalFile ^nie:dataSource $file_uri .
+} WHERE {
+    GRAPH $graph {
```

Comment:
Testing atm.
Found this bug and removed this locally, no graph. Deletes need to happen on all graphs.

Reply:
I think the query should end up like this:

```
DELETE {
    GRAPH ?g {
        $file_uri ?p ?o ;
            nie:dataSource ?logicalFile .
    }
} WHERE {
    GRAPH ?g {
        $file_uri a nfo:FileDataObject ;
            ?p ?o .
        OPTIONAL {
            ?logicalFile a nfo:FileDataObject ;
                ^nie:dataSource $file_uri .
        }
    }
}
```

`$graph` as you mention doesn't work because it's not `substitute`d in the template, but by using `?g` you ensure all the quads are selected. Also you can't do `?thing ^prop ?otherthing` inside the `DELETE` block, you have turn it around.

Reply:
not sure where my head was at when I made that query.
That `?logicalFile ^nie:dataSource $file_uri` part will just be selected when doing `$file_uri ?p ?o` 
So the optional is not required.
I thought the physical file has references to the left, but it doesn't, only to the right
so can simplify

```
DELETE {
    GRAPH ?g {
        $file_uri ?p ?o .
    }
} WHERE {
    GRAPH ?g {
        $file_uri a nfo:FileDataObject ;
            ?p ?o .
    }
}
```

Reply:
removed the graph as it doesn't seem to work.
without the data on the kanselarij graph gets removed and yggie does the rest for the other graphs